Src/VimCore/CoreInterfaces.fs

```diff
@@ -1840,15 +1855,23 @@ type BlockSpan =
         let x = x
         x.GetBlockSpansCore (fun line beforeSpaces ->
             let startColumn = SnapshotOverlapColumn.GetColumnForSpacesOrEnd(line, beforeSpaces, x.TabStop)
-            let endColumn = SnapshotOverlapColumn.GetColumnForSpacesOrEnd(line, beforeSpaces + x.SpacesLength, x.TabStop)
+            let endColumn =
+                if x.EndOfLine then
+                    SnapshotOverlapColumn(x.GetEndColumn line, x.TabStop)
+                else
+                    SnapshotOverlapColumn.GetColumnForSpacesOrEnd(line, beforeSpaces + x.SpacesLength, x.TabStop)
             SnapshotOverlapColumnSpan(startColumn, endColumn, x.TabStop))
 
     member x.BlockSpans = x.BlockColumnSpans |> NonEmptyCollectionUtil.Map (fun s -> s.Span)
 
     member x.BlockVirtualSpans = x.BlockVirtualColumnSpans |> NonEmptyCollectionUtil.Map (fun s -> s.VirtualSpan)
 
+    /// Adjust the block span for the specified 'end-of-line' setting
+    member x.AdjustForEndOfLine endOfLine =

```

Comment:
Think we should call this `WithEndOfLine`. This is essentially just changing a single property without performing any other logic / behavior under the hood. The intent of the code base was to use the naming form `With<property name>` up until now. It's a nod to F# record update expressions `{e with EndOfLine = true }`. 

Few examples (that I looked up to make sure my memory was right here): `IntervalTree` and `SnapshotOverlapColumn.WithTabStop`.